Loop/Managers/DeviceDataManager.swift

```diff
@@ -1210,6 +1211,67 @@ extension DeviceDataManager {
+                        return
+                    }
+                    
+                    guard let maxBolusAmount = loopManager.settings.maximumBolus else {
+                        NotificationManager.sendRemoteBolusFailureNotification(for: RemoteCommandError.missingMaxBolus, amount: bolusAmount)
+                        log.default("No max bolus detected. Aborting...")
+                        return
+                    }
+                    
+                    guard bolusAmount.isLessThanOrEqualTo(maxBolusAmount) else {
+                        NotificationManager.sendRemoteBolusFailureNotification(for: RemoteCommandError.exceedsMaxBolus, amount: bolusAmount)
+                        log.default("Remote bolus higher than maximum. Aborting...")
+                        return
+                    }
+                    
+                    self.enactBolus(units: bolusAmount, automatic: true) { error in
+                        if let error = error {
+                            NotificationManager.sendRemoteBolusFailureNotification(for: error, amount: bolusAmount)
+                        } else {
+                            NotificationManager.sendRemoteBolusNotification(amount: bolusAmount)
+                        }
+                    }
+                case .carbsEntry(let candidateCarbEntry):
+                    log.default("Adding carbs entry.")
+                    
+                    let candidateCarbsInGrams = candidateCarbEntry.quantity.doubleValue(for: .gram())
+                    
+                    //Remote carb entry requires validation from its remote source
+                    guard remoteDataServicesManager.validatePushNotificationSource(notification) else {
+                        NotificationManager.sendRemoteCarbEntryFailureNotification(for: RemoteCommandError.invalidOTP, amountInGrams: candidateCarbsInGrams)
+                        log.info("Could not validate notification: %{public}@", String(describing: notification))
+                        return
+                    }
+                    
+                    guard candidateCarbsInGrams > 0.0 else {
+                        NotificationManager.sendRemoteCarbEntryFailureNotification(for: RemoteCommandError.invalidCarbs, amountInGrams: candidateCarbsInGrams)
+                        log.default("Invalid carb entry amount. Aborting...")
+                        return
+                    }
+                    
+                    guard candidateCarbsInGrams <= LoopConstants.maxCarbEntryQuantity.doubleValue(for: .gram()) else {
+                        NotificationManager.sendRemoteCarbEntryFailureNotification(for: RemoteCommandError.exceedsMaxCarbs, amountInGrams: candidateCarbsInGrams)
+                        log.default("Carbs higher than maximum. Aborting...")
+                        return
+                    }
+                    
```

Comment:
Finally we attempt to add the carbs, and also show an error if that fails (unlikely - would be Core Data issue it seems)